kong_test.go

```diff
@@ -1005,9 +1005,82 @@ func TestMultipleDefaultCommands(t *testing.T) {
 		One struct{} `cmd:"" default:"1"`
 		Two struct{} `cmd:"" default:"1"`
 	}
+	_, err := kong.New(&cli)
+	require.EqualError(t, err, "<anonymous struct>.Two: can't have more than one default command under  <command>")
+}
+
+func TestDefaultCommandWithSubCommand(t *testing.T) {
+	var cli struct {
+		One struct {
+			Two struct{} `cmd:""`
+		} `cmd:"" default:"1"`
+	}
+	_, err := kong.New(&cli)
+	require.EqualError(t, err, "<anonymous struct>.One: default command one <command> must not have subcommands or arguments")
+}
+
+func TestDefaultCommandWithAllowedSubCommand(t *testing.T) {
+	var cli struct {
+		One struct {
+			Two struct{} `cmd:""`
+		} `cmd:"" default:"withargs"`
+	}
 	p := mustNew(t, &cli)
-	_, err := p.Parse([]string{})
-	require.EqualError(t, err, "can't have more than one default command under  <command>")
+	ctx, err := p.Parse([]string{"two"})
+	require.NoError(t, err)
+	require.Equal(t, "one two", ctx.Command())
+}
+
+func TestDefaultCommandWithArgument(t *testing.T) {
+	var cli struct {
+		One struct {
+			Arg string `arg:""`
+		} `cmd:"" default:"1"`
+	}
+	_, err := kong.New(&cli)
+	require.EqualError(t, err, "<anonymous struct>.One: default command one <arg> must not have subcommands or arguments")
+}
+
+func TestDefaultCommandWithAllowedArgument(t *testing.T) {
+	var cli struct {
+		One struct {
+			Arg  string `arg:""`
+			Flag string
+		} `cmd:"" default:"withargs"`
+	}
+	p := mustNew(t, &cli)
+	_, err := p.Parse([]string{"arg", "--flag=value"})
+	require.NoError(t, err)
+	require.Equal(t, "arg", cli.One.Arg)
+	require.Equal(t, "value", cli.One.Flag)
+}
+
+func TestDefaultCommandWithBranchingArgument(t *testing.T) {
+	var cli struct {
+		One struct {
+			Two struct {
+				Two string `arg:""`
+			} `arg:""`
+		} `cmd:"" default:"1"`
+	}
+	_, err := kong.New(&cli)
+	require.EqualError(t, err, "<anonymous struct>.One: default command one <command> must not have subcommands or arguments")
+}
+
+func TestDefaultCommandWithAllowedBranchingArgument(t *testing.T) {
```

Comment:
Can you add a test with multiple top-level commands, one of which has `default:"withargs"`? That is the main concern. Test that the other command does get executed. Order of fields may be important too. Commands should always have higher priority than args in this case.

Reply:
Done.